src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Created.cs

```diff
@@ -111,4 +111,73 @@ public static void FileSystemWatcher_Created_Negative()
             Utility.ExpectNoEvent(eventOccured, "created");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Created_NestedDirectories()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Created);
+
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
```

Comment:
This will work for some of them...the Deleted ones will need to stay as-is otherwise I would need to add an action to fire after every using block ended, which is basically special casing the Deleted events anyway.